wrphttp/requestResponse.go

```diff
@@ -105,16 +139,38 @@ type entityResponseWriter struct {
 	f wrp.Format
 }
 
-func (erw *entityResponseWriter) WriteWRP(v interface{}) (int, error) {
-	var (
-		output  []byte
-		encoder = wrp.NewEncoderBytes(&output, erw.f)
-	)
+func (erw *entityResponseWriter) WriteWRP(e *Entity) (int, error) {
+	if e == nil {
+		return 0, ErrUndefinedEntity
+	}
+
+	var output []byte
 
-	if err := encoder.Encode(v); err != nil {
-		return 0, err
+	if len(e.Bytes) > 0 && e.Format == erw.f {
```

Comment:
should we do a nil check of Bytes before trying to check its length?

Reply:
I added tests for that case. Len of nil byte array == 0 so we should be good